asf_heat_pump_suitability/analysis/exploratory/create_full_dataset/assign_cluster_suitability_and_feasibility.py

```diff
@@ -2,236 +2,402 @@
     """
 
-    cluster_df = cluster_df.with_columns(
-        pl.struct(["cluster_size", "hn_area", "city_centre", "has_outdoor_space"])
-        .map_elements(
-            lambda row: assign_str_suitable_techology_scheme(
-                row["cluster_size"],
-                row["hn_area"],
-                row["city_center"],
-                row["has_outdoor_space"],
+    # Transform predicted_tenure and predicted_property_type into dummies
+    df = df.to_dummies("predicted_tenure")
+    df = df.rename(
+        {
+            "predicted_tenure_owner-occupied": "owner_occupied",
+            "predicted_tenure_rental (social)": "social_housing",
+        }
+    )
+
+    df = df.to_dummies("predicted_property_type")
+    df = df.rename({"predicted_property_type_Flat, maisonette or apartment": "flats"})
+
+    # Compute on_gas
+    df = df.with_columns((~pl.col("use_off_gas")).alias("on_gas"))
+
+    # Compute not_in_listed_building
+    df = df.with_columns(
+        (~pl.col("in_listed_building")).alias("not_in_listed_building")
+    )
+    df = df.with_columns(
+        (~pl.col("in_conservation_area")).alias("not_in_conservation_area")
+    )
+
+    # Create has_outdoor_space from garden_area_m2
+    df = df.with_columns(
+        (pl.col("garden_area_m2") > outdoor_space_threshold).alias("has_outdoor_space")
+    )
+
+    # Creating close_to_city_centre from distance_to_city_centre and city_centre_threshold
+    df = df.with_columns(
+        pl.col("oa21").is_in(city_centre_oas).alias("close_to_city_centre")
+    )
+
+    # Creating imd_decile_above_avg from imd_decile
+    df = df.with_columns((pl.col("imd_decile") > 5).alias("imd_decile_above_avg"))
+
+    # Aggregating data by cluster, distance to anchor loads is calculated
+    # separately, so don't include this in this step
+
+    df = df.group_by("cluster").agg(
+        (
+            (pl.col([f for f in features if f != "close_to_anchor_loads"]).mean()).cast(
+                pl.Float64
             )
+            * 100
+        ).name.prefix("perc_"),
+        pl.col("UPRN").count().alias("cluster_size"),
+    )
+
+    # Creating close_to_anchor_loads from distance_from_anchor_property_m and anchor_loads_threshold
+
+    anchor_dist_df = anchor_dist_df.with_columns(
+        (pl.col("distance_from_anchor_property_m") <= anchor_loads_threshold).alias(
+            "close_to_anchor_loads"
         )
-        .alias("suitable_scheme")
+    ).select(
+        pl.col("cluster").cast(pl.Int64),
+        (pl.col("close_to_anchor_loads").cast(pl.Float64) * 100).name.prefix("perc_"),
     )
 
-    return cluster_df
+    df = df.join(anchor_dist_df, on="cluster")
+
+    # scale cluster size to be between 0 and 100
+    df = df.with_columns(
+        (
+            (pl.col("cluster_size") - pl.col("cluster_size").min())
+            / (pl.col("cluster_size").max() - pl.col("cluster_size").min())
+            * 100
+        ).alias("cluster_size")
+    )
```

Comment:
```suggestion
    df = df.join(anchor_dist_df, on="cluster").with_columns(
        # scale cluster size to be between 0 and 100
        (
            (pl.col("cluster_size") - pl.col("cluster_size").min())
            / (pl.col("cluster_size").max() - pl.col("cluster_size").min())
            * 100
        ).alias("cluster_size")
    )
```
I wonder if this is worth renaming as `scaled_cluster_size` so we don't get confused

Reply:
Good shout. I was also thinking that something else that would make sense to compute would be the percentage of homes in each cluster, so cluster_size divided by the total number of homes in analysis *100. do you have a preference?

Reply:
I think the scaled cluster size is better. They both kind of contain the same information but the scaled size is more intuitive to me and I think will scale better as well if we include other areas.